cdap-explore/src/test/java/co/cask/cdap/explore/service/ExploreServiceUtilsTest.java

```diff
@@ -112,4 +126,68 @@ public void hijackConfFileTest() throws Exception {
     Assert.assertEquals(confFile, ExploreServiceUtils.updateConfFileForExplore(confFile, tempDir));
   }
 
+  @Test
+  public void testRewriteHiveAuthFactory() throws Exception {
+    URL hiveAuthURL = getClass().getClassLoader().getResource(
+      HiveAuthFactory.class.getName().replace('.', '/') + ".class");
+
+    String hiveJarFilePath = hiveAuthURL.getPath();
+    File hiveJarFile = new File(URI.create(hiveJarFilePath.substring(0, hiveJarFilePath.indexOf("!/"))));
+
+    File targetJar = ExploreServiceUtils.rewriteHiveAuthFactory(hiveJarFile,
+                                                                new File(tmpFolder.newFolder(), "hive.jar"));
+
+    try (
+      TestHiveAuthFactoryClassLoader cl = new TestHiveAuthFactoryClassLoader(targetJar, getClass().getClassLoader())
+    ) {
+      Class<?> hiveAuthFactoryClass = cl.loadClass(HiveAuthFactory.class.getName());
+      Method loginFromKeytab = hiveAuthFactoryClass.getMethod("loginFromKeytab", HiveConf.class);
+
+      // If the write is not successful, this will throw exception
+      loginFromKeytab.invoke(null, new Object[] { null });
+    }
+  }
+
+  private static final class TestHiveAuthFactoryClassLoader extends ClassLoader implements Closeable {
```

Comment:
A javadoc would be helpful. Saying something like this class loader will prefer loading classes from the given jar file than the one from parent.